### PR TITLE
Converted to Swift 3 and added microphone usage description

### PR DESCRIPTION
**Pull request by dan-webb**
Converted to Swift 3.
Added NSMicrophoneUsageDescription to Info.plist, required on iOS 10.

### CODE REVIEW

File: NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift [hunk 1]
```diff
@@ -55,7 +55,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     nearbyPermission = GNSPermission(changedHandler: {[unowned self] granted in
       self.messageViewController.leftBarButton =
         UIBarButtonItem(title: String(format: "%@ Nearby", granted ? "Deny" : "Allow"),
-          style: UIBarButtonItemStyle.Bordered, target: self, action: "toggleNearbyPermission")
+          style: UIBarButtonItemStyle.plain, target: self, action: #selector(AppDelegate.toggleNearbyPermission))
```
Comment (iulukaya) [inline NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift]: nit: you can just say style: .plain

File: NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift [hunk 2]
```diff
@@ -64,22 +64,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     // Create the message manager, which lets you publish messages and subscribe to messages
     // published by nearby devices.
-    messageMgr = GNSMessageManager(APIKey: kMyAPIKey,
-      paramsBlock: {(params: GNSMessageManagerParams!) -> Void in
+    messageMgr = GNSMessageManager(apiKey: kMyAPIKey,
+      paramsBlock: {(params: GNSMessageManagerParams?) -> Void in
         // This is called when microphone permission is enabled or disabled by the user.
-        params.microphonePermissionErrorHandler = { hasError in
+        params!.microphonePermissionErrorHandler = { hasError in
```
Comment (iulukaya) [inline NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift]: force-unwrapping is bad, it's gonna crash if params is nil. You may wanna do a guard check or do an if let params = params { block.

File: NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift [hunk 3]
```diff
@@ -93,8 +93,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
   func setupStartStopButton() {
     let isSharing = (publication != nil)
     messageViewController.rightBarButton = UIBarButtonItem(title: isSharing ? "Stop" : "Start",
-      style: UIBarButtonItemStyle.Bordered,
-      target: self, action: isSharing ? "stopSharing" :  "startSharing")
+      style: UIBarButtonItemStyle.plain,
```
Comment (iulukaya) [inline NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift]: .plain

File: NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift [hunk 4]
```diff
@@ -118,23 +118,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
   /// Starts publishing the specified name and scanning for nearby devices that are publishing
   /// their names.
-  func startSharingWithName(name: String) {
+  func startSharingWithName(_ name: String) {
```
Comment (iulukaya) [inline NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift]: I'd name this startSharing(withName name: String)

File: NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift [hunk 5]
```diff
@@ -118,23 +118,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     if let messageMgr = self.messageMgr {
       // Show the name in the message view title and set up the Stop button.
       messageViewController.title = name
 
       // Publish the name to nearby devices.
-      let pubMessage: GNSMessage = GNSMessage(content: name.dataUsingEncoding(NSUTF8StringEncoding,
+      let pubMessage: GNSMessage = GNSMessage(content: name.data(using: String.Encoding.utf8,
```
Comment (iulukaya) [inline NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift]: .utf8

File: NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift [hunk 6]
```diff
@@ -118,23 +118,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
-      publication = messageMgr.publicationWithMessage(pubMessage)
+      publication = messageMgr.publication(with: pubMessage)
 
       // Subscribe to messages from nearby devices and display them in the message view.
-      subscription = messageMgr.subscriptionWithMessageFoundHandler({[unowned self] (message: GNSMessage!) -> Void in
-        self.messageViewController.addMessage(String(data: message.content, encoding:NSUTF8StringEncoding))
-        }, messageLostHandler: {[unowned self](message: GNSMessage!) -> Void in
-          self.messageViewController.removeMessage(String(data: message.content, encoding: NSUTF8StringEncoding))
+      subscription = messageMgr.subscription(messageFoundHandler: {[unowned self] (message: GNSMessage?) -> Void in
+        self.messageViewController.addMessage(String(data: message!.content, encoding:String.Encoding.utf8))
```
Comment (iulukaya) [inline NearbyMessagesExample/NearbyMessagesExampleSwift/AppDelegate.swift]: force-unwrapping

**Comment by dan-webb** [comment]
Thanks for the review, Ibrahim!  I've added a couple of commits for the changes you requested, and they include some updates to the Podfile syntax also.

However, I'm not sure if my commits are showing up here.  Can you see them in the diffs?  Let me know if there's something I need to do to push them here.

**Pull request merged**